runtime/src/iree/vm/ops.h

```diff
@@ -569,10 +569,18 @@ static inline float vm_round_f32_even(float operand) {
 #endif  // C23
 }
 static inline float vm_min_f32(float lhs, float rhs) {
-  return rhs < lhs ? rhs : lhs;
+  // If `lhs` is NaN, returns `rhs`.
+  if (lhs != lhs) return rhs;
+  // If `rhs` is NaN, returns `lhs`.
+  if (rhs != rhs) return lhs;
+  return lhs < rhs ? lhs : rhs;
 }
```

Comment:
Do we care about the NaN handling for a fully compliant lowering here? I'd expect well-behaved / numerically stable programs to want the skip the extra branches.

Reply:
Good point. It does generates two more instructions on x86 machines. E.g., `std::min` becomes `minss`, and `fmin` becomes `ucomisd + jp + minsd`. The change is driven by Ben's comment: https://github.com/iree-org/iree/issues/17779#issuecomment-2200575337

> good q - IMO we only need one, and the more sane behavior (maxnumf/minnumf) seems ideal

IMO, it is not critical in terms of performance because VM is a very lightweight interpreter. On the VMVX side, I don't expect people to run big programs with it. So it is fine to me.

If we care about the distinction, there are more work in plumbing through. I can follow https://github.com/iree-org/iree/commit/459f5ffad7b33e3bebd2947df8efa9cccea67b75 to add the support.

The main motivation of the PR is dropping the LLVM revert. If there is an idea about how it should be done, I'm happy to implement it or send a follow-up.